From 373e49960f8b3484cc4c1f94c3a8e7ce9f6bce93 Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Thu, 10 Jul 2014 01:03:45 -0400
Subject: [acpi] dock: fix error return code in dock_add()

Message-id: <20140710010345.22146.54263.stgit@gir.stowe>
Patchwork-id: 85592
O-Subject: [RHEL7.1 PATCH 053/105] ACPI / dock: fix error return code in dock_add()
Bugzilla: 1114228
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1114228
Upstream Status: 0177f29fea534ef5e6af2d76e9a9be0fdd325c4d

commit 0177f29fea534ef5e6af2d76e9a9be0fdd325c4d
Author: Wei Yongjun <yongjun_wei@trendmicro.com.cn>
Date:   Wed Jul 17 08:33:25 2013 +0800

    ACPI / dock: fix error return code in dock_add()

    Fix to return -ENODEV in the acpi notify handler install error
    handling case instead of 0, as done elsewhere in this function.

    Signed-off-by: Wei Yongjun <yongjun_wei@trendmicro.com.cn>
    Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/acpi/dock.c b/drivers/acpi/dock.c
index f601658..b527c1b 100644
--- a/drivers/acpi/dock.c
+++ b/drivers/acpi/dock.c
@@ -881,8 +881,10 @@ static int __init dock_add(acpi_handle handle)
 
  status = acpi_install_notify_handler(handle, ACPI_SYSTEM_NOTIFY,
           dock_notify_handler, dock_station);
- if (ACPI_FAILURE(status))
+ if (ACPI_FAILURE(status)) {
+  ret = -ENODEV;
   goto err_rmgroup;
+ }
 
  dock_station_count++;
  list_add(&dock_station->sibling, &dock_stations);
-- 
1.7.1